web/libs/ui/src/lib/button/button.tsx

```diff
@@ -0,0 +1,113 @@
+import { cn } from "../../utils/utils";
+import type { ButtonHTMLAttributes, PropsWithChildren, ReactNode } from "react";
+import styles from "./button.module.scss";
+
+const variants = {
+  primary: styles["variant-primary"],
+  neutral: styles["variant-neutral"],
+  negative: styles["variant-negative"],
+  positive: styles["variant-positive"],
+  warning: styles["variant-warning"],
+  inverted: styles["variant-neutral-interted"],
+};
+
+const looks = {
+  filled: styles["look-filled"],
+  string: styles["look-string"],
+  outlined: styles["look-outlined"],
+};
+
+const sizes = {
+  medium: styles["size-medium"], // 40px
+  small: styles["size-small"], // 32px
+  smaller: styles["size-smaller"], // 24px
+};
+
+const alignment = {
+  default: styles["align-default"],
+  center: styles["align-center"],
+  left: styles["align-left"],
+  right: styles["align-right"],
+};
+
+/**
+ * Generates a className string with button styling that can be applied to any element
+ *
+ * This utility function creates a consistent button styling that can be applied not only to
+ * button elements but also to other interactive elements like links (`<a>` tags), divs, or spans
+ * that need to visually appear as buttons.
+ *
+ * @example
+ * // Apply button styling to a link
+ * <a href="/path" className={buttonVariant({ variant: 'primary', look: 'outlined' })}>
+ *   Link that looks like a button
+ * </a>
+ */
+export function buttonVariant(
+  {
+    variant = "primary",
+    look = "filled",
+    size = "medium",
+    align = "default",
+    waiting = false,
+  }: {
+    variant?: ButtonProps["variant"];
+    look?: ButtonProps["look"];
+    size?: ButtonProps["size"];
+    align?: ButtonProps["align"];
+    waiting?: boolean;
+  },
+  className?: string,
+) {
+  const buttonStyles = [styles.base, variants[variant], looks[look], sizes[size], alignment[align]];
+  return cn(
+    "inline-flex items-center rounded-smaller border text-shadow-button p-tight box-border border transition-all",
```

Comment:
are these all tokens we should be using from figma? specifically `p-tight` caught my attention

Reply:
`tight` is a spacing tailwind class for the token for `--spacing-tight: 0.5rem` that comes from our design tokens

Reply:
perfect, i wasn't sure if that was one of ours or a native tailwind one

Reply:
yup, all from tokens. everything found in the storybook :)